app/jobs/reports/monthly_key_metrics_report.rb

```diff
@@ -6,47 +6,42 @@ class MonthlyKeyMetricsReport < BaseReport
 
     attr_reader :report_date
 
+    Report = Struct.new(:metadata, :table, :csv_name)
+
     def perform(date = Time.zone.today)
       @report_date = date
 
-      account_reuse_table = account_reuse_report.account_reuse_report
-      total_profiles_table = account_reuse_report.total_identities_report
-      account_deletion_rate_table = account_deletion_rate_report.account_deletion_report
-      total_user_count_table = total_user_count_report.total_user_count_report
-
-      upload_to_s3(account_reuse_table, report_name: 'account_reuse')
-      upload_to_s3(total_profiles_table, report_name: 'total_profiles')
-      upload_to_s3(account_deletion_rate_table, report_name: 'account_deletion_rate')
-      upload_to_s3(total_user_count_table, report_name: 'total_user_count')
+      reports = []
+      reports << Report.new(
+        metadata: account_reuse_report.account_reuse_report_metadata,
+        table: account_reuse_report.account_reuse_report,
+        csv_name: 'account_reuse'
+      )
+      reports << Report.new(
+        metadata: account_reuse_report.total_identities_report_metadata,
+        table: account_reuse_report.total_identities_report,
+        csv_name: 'total_profiles'
+      )
+      reports << Report.new(
+        metadata: account_deletion_rate_report.account_deletion_report_metadata,
+        table: account_deletion_rate_report.account_deletion_report,
+        csv_name: 'account_deletion_rate'
+      )
+      reports << Report.new(
+        metadata: total_user_count_report.total_user_count_report_metadata,
+        table: total_user_count_report.total_user_count_report,
+        csv_name: 'total_user_count'
+      )
 
-      email_tables = [
-        [
-          {
-            title: "IDV app reuse rate #{account_reuse_report.stats_month}",
-            float_as_percent: true,
-            precision: 4,
-          },
-          *account_reuse_table,
-        ],
-        [
-          { title: 'Total proofed identities' },
-          *total_profiles_table,
-        ],
-        [
-          {
-            title: 'Account deletion rate (last 30 days)',
-            float_as_percent: true,
-            precision: 4,
-          },
-          *account_deletion_rate_table,
-        ],
-        [
-          {
-            title: 'Total user count (all-time)',
-          },
-          *total_user_count_table,
-        ],
-      ]
+      email_tables = []
+      reports.each do |report|
+        upload_to_s3(report.table, report_name: report.csv_name)
+        
+        email_tables << [
+          report.metadata,
+          *report.table
+        ]
+      end
```

Comment:
```suggestion
      reports.each do |report|
        upload_to_s3(report.table, report_name: report.csv_name)
      end

      email_tables = reports.map do |report|        
        [
          report.metadata,
          *report.table
        ]
      end
```

Reply:
At the time I wrote this I combined these into one loop because it would be more efficient, but seeing what you propose, I think the "inefficiency" of going through a loop of what may one day grow to 10 things twice is worth being much easier to read. :)